src/Access/AccessControl.cpp

```diff
@@ -651,6 +652,31 @@ bool AccessControl::isPlaintextPasswordAllowed() const
     return allow_plaintext_password;
 }
 
+void AccessControl::setDefaultPasswordTypeFromConfig(const String & type_)
+{
+    if (type_ == "plaintext")
+        default_password_type = AuthenticationType::PLAINTEXT_PASSWORD;
+    else if (type_ == "double_sha1")
+        default_password_type = AuthenticationType::DOUBLE_SHA1_PASSWORD;
+    else if (type_ == "sha256")
+        default_password_type = AuthenticationType::SHA256_PASSWORD;
+    else
+        throw Exception("Unknown password type in 'default_password_type' in config", ErrorCodes::UNKNOWN_ELEMENT_IN_CONFIG);
+}
+
+void AccessControl::setDefaultPasswordType(const AuthenticationType & type_)
+{
+    if (type_ < AuthenticationType::MAX)
+        default_password_type = type_;
+    else
+        default_password_type = AuthenticationType::SHA256_PASSWORD;
```

Comment:
In this case the function should throw an exception, not just silently convert a wrong value to sha256.

Reply:
This function is only used when the config parameter is already parsed and assigned a numeric value from enum, so it should be correct. There can be a situation (in the future) where we added a new password type and it does not yet exists in the client version. In this case, I'm not sure whether the client should throw an exception that the chosen default method is not supported and that we would use SHA256.

Reply:
But in general I think I will change it to something like this:
```
    if (type_ == AuthenticationType::PLAINTEXT_PASSWORD ||
         type_ == AuthenticationType::DOUBLE_SHA1_PASSWORD ||
         type_ == AuthenticationType::SHA256_PASSWORD)
    {
        default_password_type = type_;
    }
    else
    {
        default_password_type = AuthenticationType::SHA256_PASSWORD;
        throw Exception("'default_password_type' in config is not supported in this client version, SHA256_PASSWORD will be used", ErrorCodes::BAD_ARGUMENTS);
    }
```